### DISCUSSION TITLE
One Little Thing: Embed a File in the HTML Output of R Markdown - Want to embed data or source files in HTML output? No problem! | /en/2018/07/embed-file/

### DISCUSSION THREAD

**Discussion by yihui** [rbind/yihui#1203] [Comments]
https://yihui.org/en/2018/07/embed-file/

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
DId you read the help page ?xfun::embed_files and install Rtools? You can verify it via Sys.which('zip').

> Originally posted on 2018-09-10 20:23:58

**Reply by yihui** [reply]
It was partially my fault. The error message was confusing. I have provided a more informative one after seeing your message: <https://github.com/yihui/xfun/commit/dc8eb19d3ae0555558306a0af25aa9350533d95a>

> Originally posted on 2018-09-11 15:47:04

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
Like this?  

    embed_var = function(x, ...) {
      f = tempfile(fileext = '.rds')
      saveRDS(x, f)
      xfun::embed_file(f, ...)
    }

> Originally posted on 2019-02-02 15:48:47

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
This function was added in xfun v0.4 and released to CRAN ten months ago: <https://github.com/yihui/xfun/releases> I don't use conda, and I don't know how to update R packages in it. In R, you just run update.packages().

> Originally posted on 2019-08-30 14:38:00

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
I cannot reproduce this issue with either xfun::embed_files("test.html") or xfun::embed_files("test.html") and Chrome on Ubuntu 18.04. Could you file an issue to <https://github.com/yihui/xfun> and upload the problematic example (you can zip all files and upload in the issue)?

BTW, please make sure your issue is different with the existing one: <https://github.com/yihui/xfun/issues/23> Thanks!

> Originally posted on 2020-01-16 20:22:56

[a comment by an automated account is not shown]

**Comment by jenineharris** [comment]
Is there any way to embed a password so that, when people click on the link to download, they are prompted for a password first? I read through the help documentation and did some searching but did not see this functionality. Thanks for such a great package!!!

**Reply by yihui** [reply]
Password protection is likely to be meaningless to people who understand HTML and base64 encoding, because the file is embedded in HTML as a base64 string. As long as you give them the HTML page, they will have access to this string, and they can easily decode it by themselves without clicking on the link. If you want to password protect a file, I'd recommend encrypting it before embedding it. Then people can download and decrypt it using the password you told them.

That said, it's certainly possible to prompt a password input---it just won't really protect your file, and people can easily work around it.

**Reply by jenineharris** [reply]
Ok, thank you! That makes sense.
